### PR TITLE
Add long error explanation for E0634

### PR DESCRIPTION
**Pull request by ayushmishra2005**
Add long explanation for the E0634 error code
Part of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0634.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+A type has conflicting `packed` representation hint.
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0634.md]:
```suggestion
A type has conflicting `packed` representation hints.
```

File: src/librustc_error_codes/error_codes/E0634.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+```compile_fail,E0634
+#[repr(packed, packed(2))] // error!
+struct Company(i32);
+
+#[repr(packed(2))] // error!
+#[repr(packed)]
+struct Company(i32);
+```
+
+You cannot use conflicting `packed` hint on a same type. If you want to pack a
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0634.md]:
```suggestion
You cannot use conflicting `packed` hints on a same type. If you want to pack a
```

**Review by Dylan-DPC-zz — CHANGES_REQUESTED** [review]

**Review by GuillaumeGomez — APPROVED** [review]
Looks good to me (minus the corrections @Dylan-DPC asked for)!

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r=Dylan-DPC,GuillaumeGomez rollup

**Pull request merged**